internal/scanner/walk.go

```diff
@@ -97,6 +101,9 @@ func (w *Walker) Walk() (chan protocol.FileInfo, error) {
 	newParallelHasher(w.Dir, w.BlockSize, w.Hashers, hashedFiles, files)
 
 	go func() {
+		w.symlinksSeen = make(map[string]struct{})
+		w.dirsToScan = nil
```

Comment:
Yeah, it's ugly. We can't affect the return value of filepath.Walk so our options are a closure over the walk func or an attribute like now. Probably the closure is better, as we already do that for the channel and so on. I had a different thing going on at the beginning. :)